src/routes/(console)/organization-[organization]/billing/retryPaymentModal.svelte

```diff
@@ -71,18 +71,21 @@
             if (setAsDefault) {
                 await sdk.forConsole.billing.setDefaultPaymentMethod(paymentMethodId);
             }
-            const { clientSecret } = await sdk.forConsole.billing.retryPayment(
+            const { clientSecret, status } = await sdk.forConsole.billing.retryPayment(
                 $organization.$id,
                 invoice.$id,
                 paymentMethodId
             );
 
-            await confirmPayment(
-                $organization.$id,
-                clientSecret,
-                paymentMethodId ? paymentMethodId : $organization.paymentMethodId,
-                `${base}/organization-${$organization.$id}/billing?type=validate-invoice&invoice=${invoice.$id}`
-            );
+            if (status !== 'succeeded' && status !== 'cancelled') {
+                // probably still pending, confirm via stripe!
+                await confirmPayment(
+                    $organization.$id,
+                    clientSecret,
+                    paymentMethodId ? paymentMethodId : $organization.paymentMethodId,
+                    `${base}/organization-${$organization.$id}/billing?type=validate-invoice&invoice=${invoice.$id}`
+                );
+            }
```

Comment:
may be we need a success notification something `invoice successfully processed` in else condition?

Reply:
For both, the notification should be same I think. If its processed, we show - `Payment has been successfully processed`, if not, we confirm and show the same anyways. Makes sense? Can change `Payment` to `Invoice`.